tests/acceptance/dns_test.go

```diff
@@ -137,6 +165,51 @@ var _ = Describe("DNS", func() {
 		Expect(records).To(HaveLen(1))
 		Expect(records[0].String()).To(Equal("10.0.0.1"))
 
+		By("creating an A record for the bastion1")
+		Eventually(func() error {
+			var err error
+			records, err = resolver.LookupIP(ctx, "ip", bastionDomain)
+			return err
+		}).Should(Succeed())
+
+		Expect(records).To(HaveLen(1))
+		Expect(records[0].String()).To(Equal("1.2.3.4"))
+
+		By("updating an A record for the bastion1")
+		// update bastion machine with different IP
+		patchedMachine := machine.DeepCopy()
+		patchedMachine.Status.Addresses = []corev1.NodeAddress{
+			{
+				Type:    "ExternalIP",
+				Address: "1.2.3.5",
+			},
+		}
+		Expect(k8sClient.Status().Patch(ctx, patchedMachine, client.MergeFrom(machine))).To(Succeed())
+
+		// trigger reconciliation loop by update to speed up tests
+		patchedCluster := gcpCluster.DeepCopy()
+		patchedCluster.Labels = map[string]string{
+			"extra": "label",
+		}
+		Expect(k8sClient.Status().Patch(ctx, patchedCluster, client.MergeFrom(gcpCluster))).To(Succeed())
+
+		Eventually(func() error {
+			var err error
+			records, err = resolver.LookupIP(ctx, "ip", bastionDomain)
+			if err != nil {
+				return err
+			}
+			// we need to wait until the record is updated
+			if records[0].String() == "1.2.3.4" {
+				return errors.New("record not updated yet")
+			}
```

Comment:
nitpick: It can be written like this too ([Eventually docs Category 3](https://pkg.go.dev/github.com/onsi/gomega#Eventually)):
```suggestion
		
		Eventually(func(g Gomega) (string, error) {
			var err error
			records, err = resolver.LookupIP(ctx, "ip", bastionDomain)
			if err != nil {
				return "", err
			}

			g.Expect(records).To(HaveLen(1))
			return records[0].String(), nil
		}).Should(Equal("1.2.3.5"))

```